### DISCUSSION TITLE
thinking of adding Maybe.ToUnitResult() extension

### DISCUSSION THREAD

**Discussion by julienasp** [vkhorikov/CSharpFunctionalExtensions#542] [Ideas]
Hi everyone, 

i just had a case at work, where i had a validation method (in the domain), that was returning a Maybe<Error>, i use that validation service in my applicative service, and there's no error mapping into a more generic one, then i return a  UnitResult<ApplicationError>.

so the code is very much like:

```
var result = _domainService.Validate();
result.Map(MapToApplicationError);

return result.HasValue ? UnitResult.Failure(result.Value) : UnitResult.Success<ApplicationError>();
```

but i was thinking of something like this would be great to have:

```
return _domainService.Validate()
.Map(MapToApplicationError)
.ToUnitResult();
```
 
 i was just wondering what you guys think, before submitting this.
 
 so the behaviour would be 
 if theres a value in the maybe -> ToUnitResult -> UnitResult failure
 if theren no value in the value -> toUnitResult -> UnitResult Success.

**Comment by julienasp** [comment]
After reading ToResult()

i decided to change my logic, to match the logic that ToResult exposed, otherwise it would be too confusing

**Reply by julienasp** [reply]
in fact, i added both, if an error is provided, we match the logic that ToResult has, otherwise, we fallback with what my original post mentionnel. Maybe<Error> -> Has value -> Failure

**Comment by vkhorikov** [comment]
I'd personally recommend using `Result<Unit, T>` instead of `UnitResult`. From my experience, it becomes to cumbersome to maintain multiple variations of `Result` in the app.

As for the extension method itself, we can definitely add it to the library, feel free to submit a PR.
